authenticatorbridge/src/test/java/com/bitwarden/authenticatorbridge/manager/AuthenticatorBridgeManagerTest.kt

```diff
@@ -0,0 +1,440 @@
+                SYMMETRIC_KEY.toFingerprint().getOrNull()
+            )
+        } returns true
+        fakeSymmetricKeyStorageProvider.symmetricKey = SYMMETRIC_KEY
+        val serviceConnection = slot<ServiceConnection>()
+        val callback = slot<IAuthenticatorBridgeServiceCallback>()
+        val mockIntent: Intent = mockk()
+        every { IAuthenticatorBridgeService.Stub.asInterface(any()) } returns mockBridgeService
+        every { mockBridgeService.registerBridgeServiceCallback(capture(callback)) } just runs
+        every {
+            anyConstructed<Intent>().setComponent(any())
+        } returns mockIntent
+        every { context.unbindService(any()) } just runs
+        every {
+            context.bindService(
+                any(),
+                capture(serviceConnection),
+                Context.BIND_AUTO_CREATE
+            )
+        } returns true
+
+        fakeLifecycleOwner.lifecycle.dispatchOnStart()
+        serviceConnection.captured.onServiceConnected(mockk(), mockk())
+        assertEquals(AccountSyncState.Loading, manager.accountSyncStateFlow.value)
+
+        callback.captured.onAccountsSync(encryptedAccounts)
+        assertEquals(AccountSyncState.Success(expectedAccounts), manager.accountSyncStateFlow.value)
+
+        assertEquals(fakeSymmetricKeyStorageProvider.symmetricKey, SYMMETRIC_KEY)
+        verify(exactly = 0) { mockBridgeService.symmetricEncryptionKeyData }
+        verify { context.bindService(any(), any(), Context.BIND_AUTO_CREATE) }
+        verify { mockBridgeService.registerBridgeServiceCallback(any()) }
+        verify { mockBridgeService.syncAccounts() }
+    }
+
+    @Test
+    fun `onAccountsSync when symmetric key is missing should not set state to Success`() {
+        val encryptedAccounts: EncryptedSharedAccountData = mockk()
+        every {
+            mockBridgeService.checkSymmetricEncryptionKeyFingerprint(
+                SYMMETRIC_KEY.toFingerprint().getOrNull()
+            )
+        } returns true
+        fakeSymmetricKeyStorageProvider.symmetricKey = SYMMETRIC_KEY
+        val serviceConnection = slot<ServiceConnection>()
+        val callback = slot<IAuthenticatorBridgeServiceCallback>()
+        val mockIntent: Intent = mockk()
+        every { IAuthenticatorBridgeService.Stub.asInterface(any()) } returns mockBridgeService
+        every { mockBridgeService.registerBridgeServiceCallback(capture(callback)) } just runs
+        every {
+            anyConstructed<Intent>().setComponent(any())
+        } returns mockIntent
+        every { context.unbindService(any()) } just runs
+        every {
+            context.bindService(
+                any(),
+                capture(serviceConnection),
+                Context.BIND_AUTO_CREATE
+            )
+        } returns true
+
+        fakeLifecycleOwner.lifecycle.dispatchOnStart()
+        serviceConnection.captured.onServiceConnected(mockk(), mockk())
```

Comment:
Can we handle this just by doing an `answers` where we trigger this?

```kotlin
 every { mockBridgeService.registerBridgeServiceCallback(capture(callback)) } answers {
    (it.invocation.args.first() as IAuthenticatorBridgeServiceCallback).onAccountsSync(encryptedAccounts)
}
```

Just keeps it in one place.